src/composition/pipeline.rs

```diff
@@ -134,11 +138,27 @@ impl CompositionPipeline<state::ResolveFederationVersion> {
                 &SubgraphPrompt::default(),
             )
             .await?;
-        let federation_version = federation_version.unwrap_or_else(|| {
+        let fed_two_subgraphs = fully_resolved_supergraph_config
+            .subgraphs()
+            .iter()
+            .filter_map(|(name, subgraph)| {
+                if subgraph.is_fed_two {
+                    Some(name.clone())
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<String>>();
+        let federation_version = if let Some(fed_version) = federation_version {
+            if !fed_two_subgraphs.is_empty() && fed_version.is_fed_one() {
+                return Err(FederationOneWithFederationTwoSubgraphs(fed_two_subgraphs));
+            }
+            fed_version
+        } else {
```

Comment:
We scan quite early for the mismatch in Fed Versions so we can bail out early on